### PR TITLE
diagnostics: promote from openshift ex to oadm

### PR DESCRIPTION
**Pull request by sosiouxme**
updated cmd/e2e tests to match
"openshift ex diagnostics" still works but has small deprecation notice
pointing to "oadm diagnostics"; "openshift ex validate" deprecation
notice also changed to refer to same

### CODE REVIEW

**Comment by sosiouxme** [comment]
fixes https://github.com/openshift/origin/issues/6276

@smarterclayton PTAL

**Comment by sosiouxme** [comment]
[test]

File: pkg/cmd/admin/diagnostics/diagnostics.go [hunk 1]
```diff
@@ -80,13 +82,23 @@ you will receive an error if they are not found. For example:
...
+	deprecationNotice = `use "oadm %[1]s" to run diagnostics instead.
+**************************************************************************
+The "%[2]s" command is deprecated and will be removed.
+**************************************************************************
 `
+)
+
+func NewCmdExDiagnostics(name string, fullName string, out io.Writer) *cobra.Command {
+	cmd := NewCmdDiagnostics(name, fullName, out)
+	cmd.Deprecated = fmt.Sprintf(deprecationNotice, DiagnosticsRecommendedName, fullName)
```
Comment (deads2k) [inline pkg/cmd/admin/diagnostics/diagnostics.go]: I'd prefer to do this at point of use rather have a one off method here.

File: pkg/cmd/admin/diagnostics/diagnostics.go [hunk 2]
```diff
@@ -80,13 +82,23 @@ you will receive an error if they are not found. For example:
 Diagnostics may be individually run by passing diagnostic name as arguments.
 The available diagnostic names are:
 %[2]s
+`
 
-NOTE: This is a beta version of diagnostics and may still evolve in a
-different direction.
+	deprecationNotice = `use "oadm %[1]s" to run diagnostics instead.
```
Comment (deads2k) [inline pkg/cmd/admin/diagnostics/diagnostics.go]: This line is sufficient.  If you'd like to add something about removal that's fine, but no eye catchers.

File: pkg/cmd/admin/validate/validate.go [hunk 3]
```diff
@@ -16,8 +16,9 @@ const (
 The commands here allow administrators to validate the integrity of configuration files.
 `
 
-	validateDeprecationMessage = `The commands here are deprecated and will be removed. Use 'openshift ex diagnostics' to run configuration validations instead.
-See sub-command help text for specific instructions with 'openshift ex diagnostics'.`
+	validateDeprecationMessage = `and will be removed.
```
Comment (deads2k) [inline pkg/cmd/admin/validate/validate.go]: Looks like you missed the master and node variants.  Also, two lines was already pushing it, don't add a third.

File: test/cmd/diagnostics.sh [hunk 4]
```diff
@@ -16,15 +16,15 @@ os::log::install_errexit
 # Without things feeding into systemd, AnalyzeLogs and UnitStatus are irrelevant.
 # The rest should be included in some fashion.
 
-os::cmd::expect_success 'openshift ex diagnostics ClusterRoleBindings ClusterRoles ConfigContexts '
+os::cmd::expect_success 'oadm diagnostics ClusterRoleBindings ClusterRoles ConfigContexts '
```
Comment (deads2k) [inline test/cmd/diagnostics.sh]: Add a specific check to make sure that we don't remove `openshift ex diagnostics` by accident and add a comment to keep people from removing it until we're ready.

**Comment by deads2k** [comment]
Minor comments, then lgtm

File: pkg/cmd/admin/diagnostics/diagnostics.go (hunk 1 above)
Comment (sosiouxme) [inline pkg/cmd/admin/diagnostics/diagnostics.go]: On Wed, Feb 24, 2016 at 12:43 PM, David Eads <notifications@github.com>
wrote:
>
> I'd prefer to do this at point of use rather have a one off method here.
>
>
Sure, makes sense.

File: pkg/cmd/admin/validate/validate.go (hunk 3 above)
Comment (sosiouxme) [inline pkg/cmd/admin/validate/validate.go]: I did miss those, thanks.

File: test/cmd/diagnostics.sh (hunk 4 above)
Comment (sosiouxme) [inline test/cmd/diagnostics.sh]: On Wed, Feb 24, 2016 at 12:50 PM, David Eads <notifications@github.com>
wrote:
>
> Add a specific check to make sure that we don't remove openshift ex
> diagnostics by accident and add a comment to keep people from removing it
> until we're ready.
>
>
>
done

**Comment by deads2k** [comment]
lgtm, squash and merge at will.

**Comment by sosiouxme** [comment]
[merge]

**Comment by sosiouxme** [comment]
https://ci.openshift.redhat.com/jenkins/job/merge_pull_requests_origin/5102/ looks like a flake https://github.com/openshift/origin/issues/7616

[merge] again plz

**Comment by sosiouxme** [comment]
https://ci.openshift.redhat.com/jenkins/job/merge_pull_requests_origin/5104/ looks like etcd flake https://github.com/openshift/origin/issues/7243

Third time's a charm... [merge]

**Pull request merged**